### PR TITLE
Add lock compatibility to all Building_Doors

### PR DESCRIPTION
**Pull request by JonnyNova**
Adds support for other mods adding doors based on `Building_Door`

### CODE REVIEW

**Comment by JonnyNova** [comment]
Any chance at getting this merged?

**Comment by Aviuz** [comment]
it is from the beginning
am I wrong?

**Comment by JonnyNova** [comment]
The existing implementation is fine for the base Rimworld doors, but it won't pick up mod doors that inherit from building door. The patch filters all defs by `Building_Door` class instead to ensure all doors that can be supported are patched.

**Comment by Aviuz** [comment]
sorry I missed that

**Comment by JonnyNova** [comment]
No worries!

**Comment by Aviuz** [comment]
it will only add ITab, but yes I will add this anyway

**Comment by Aviuz** [comment]
just I will leave it in base method, because you changed name and location of it dont know why

**Comment by JonnyNova** [comment]
It also has a change to the csproj to output the built binary directly into the right assemblies folder

**Comment by Aviuz** [comment]
ehm what?

**Comment by Aviuz** [comment]
all binaries are automated and works well for me

File: Source/Locks.csproj [hunk 1]
```diff
@@ -25,7 +25,7 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
+    <OutputPath>..\Assemblies\</OutputPath>
```
Comment (JonnyNova) [inline Source/Locks.csproj]: Puts the .dll in the right spot without having to copy it

Comment (Aviuz) [reply Source/Locks.csproj]: oh I don't use release anyway

Comment (Aviuz) [reply Source/Locks.csproj]: yea but in debug it works

**Comment by Aviuz** [comment]
https://github.com/Aviuz/Locks/commit/1c44bbcf8e6aa91e86469d174b0bbe5426eef755

I've pushed your changes.
I've kept only some changes, because rest of them I didn't liked (e.g. moving this to new class)

**Comment by Aviuz** [comment]
@JonnyNova 
... and sorry once again for such late reply, but I somehow forget about it

**Comment by JonnyNova** [comment]
No worries, glad it was useful

**Pull request closed without merging**